### ISSUE TITLE
from metrics import AverageMeter, Result

### ISSUE THREAD

**Issue by gentlemanman** [dontLoveBugs/DORN_pytorch#2]
"from metrics import AverageMeter, Result" in NYUDepth_train.py is a error, i can not understand

**Comment by gentlemanman** [comment]
In DORN_nyu.py OrdinalRegressionLayer , can you explain the code below
       `A = x[:, ::2, :, :].clone()
        B = x[:, 1::2, :, :].clone()
        A = A.view(N, 1, ord_num * H * W)
        B = B.view(N, 1, ord_num * H * W)
        C = torch.cat((A, B), dim=1)             # (b, 2, 6169028)
        # C = torch.clamp(C, min = 1e-8, max = 1e8) # prevent nans
        ord_c = nn.functional.softmax(C, dim=1)  # (b, 2, 6169028)
        ord_c1 = ord_c[:, 1, :].clone()          # (b, 1, 6169028)
        ord_c1 = ord_c1.view(-1, ord_num, H, W)  # (b, 68, 257, 353)
        decode_c = torch.sum(ord_c1, dim=1).view(-1, 1, H, W) #(b, 1, 257, 353)
        return decode_c, ord_c1`

**Comment by gentlemanman** [comment]
In DORN_nyu.py OrdinalRegressionLayer, what's the meaning of `ord_c1 = ord_c[:, 1, :].clone() `.  I haven’t understood it for a long time.

**Comment by dontLoveBugs** [comment]
Hi, you can see Sec3.3 in the paper, the p^k_(w, h) is computed as follows:
![image](https://user-images.githubusercontent.com/17598076/50480857-7aab0800-0a19-11e9-915f-304fc57e4240.png)
The "ord_c"  contains e^y(w, h, 2k) / (e^y(w, h, 2k) + e^(y, w, 2k+1)) and e^(w, h, 2k+1) / (e^y(w, h, 2k) + e^(w, h, 2k+1)). The ord_c[:, 1, :] is p^k_(w, h).
